### PR TITLE
feat(manage): log and show game/achievement badge changes

### PR DESCRIPTION
**Pull request by wescopeland**
This PR enhances the management app so that game and achievement badge changes are now logged and shown in the game/achievement audit log pages.

Screenshot 2024-08-16 at 5 34 18 PM

Screenshot 2024-08-16 at 5 28 52 PM

### CODE REVIEW

File: resources/views/filament/pages/audit-log.blade.php [hunk 1]
```diff
@@ -48,56 +48,65 @@
...
                                             <div class="inline-block">
                                                 <x-filament::badge>
                                                     {{ $relatedModel->name }}
                                                 </x-filament::badge>
                                                 {{ collect($oldValue)->where('id', $relatedModel->name)->get('attributes') }}
                                             </div>
                                         @endforeach
-                                    @elseif(is_array($oldValue))
+                                    @elseif ($this->getIsImageField($field))
+                                        <img src="{{ $oldValue }}" alt="Old Image" class="max-w-full h-auto"/>
```
Comment (wescopeland) [inline resources/views/filament/pages/audit-log.blade.php:85]: This file's diff is mostly just whitespace changes. Lines 84-85 and 107-108 are the actual logic changes in the file.

[the remark this answers is not in the document]
Comment (Jamiras) [reply resources/views/filament/pages/audit-log.blade.php:85]: Should the field label still be "Badge Name"?

[the remark this answers is not in the document]
Comment (wescopeland) [reply resources/views/filament/pages/audit-log.blade.php:85]: Fixed both issues in latest.

The label for both fields is now simply "Badge". Additionally, if the value is null, the page won't try to render the image anymore.

File: app/Filament/Pages/ResourceAuditLog.php [hunk 2]
```diff
@@ -127,11 +127,37 @@ protected function transformFieldValues(array $values): array
...
         }
 
         return $values;
     }
 
+    protected function getIsImageField(string $fieldName): bool
+    {
+        return in_array($fieldName, [
+            'BadgeName',
+            'ImageIcon',
```
Comment (Jamiras) [inline app/Filament/Pages/ResourceAuditLog.php:143]: We aren't currently tracking changes to Title/InGame/BoxArt. I'm assuming those would be added here at that point?

**Review by Jamiras — APPROVED** [review]

File: app/Filament/Pages/ResourceAuditLog.php (hunk 2 above)
Comment (wescopeland) [reply app/Filament/Pages/ResourceAuditLog.php:143]: They probably should be, but could need some more specialized CSS concerns given how variable the dimensions on those assets tend to be. BadgeName and ImageIcon felt like easier & valuable targets to pick off for the time being.

**Pull request merged**